### PR TITLE
add readonly address

### PR DESCRIPTION
**Pull request by shileiwill**
AUTO-8803

- yarn prettier:write
- yarn hardhat size-contracts
- pnpm hardhat test test/v0.8/automation/AutomationRegistry2_2.test.ts all tests passed.
- pnpm ts-node ./scripts/generate-automation-master-interface.ts nothing changed.
- go generate core/gethwrappers/go_generate.go some wrappers regenerated.
- `forge test --match-path "*/AutomationRegistry2_2.t.sol" ` all foundry tests pass.

### CODE REVIEW

File: contracts/src/v0.8/automation/dev/test/AutomationRegistry2_2.t.sol [hunk 1]
```diff
@@ -13,6 +13,7 @@ contract AutomationRegistry2_2_SetUp is BaseTest {
   address internal constant LINK_ETH_FEED = 0x1111111111111111111111111111111111111110;
   address internal constant FAST_GAS_FEED = 0x1111111111111111111111111111111111111112;
   address internal constant LINK_TOKEN = 0x1111111111111111111111111111111111111113;
+  address internal constant ZERO_ADDRESS = 0x0000000000000000000000000000000000000000;
```
Comment (FelixFan1992) [inline contracts/src/v0.8/automation/dev/test/AutomationRegistry2_2.t.sol:16]: try `address(0)`

File: contracts/src/v0.8/automation/dev/test/AutomationRegistry2_2.t.sol
Comment (FelixFan1992) [inline contracts/src/v0.8/automation/dev/test/AutomationRegistry2_2.t.sol:1]: let's add a test here to expect a revert if someone sends a eth_call with different tx.origin

Comment (shileiwill) [reply contracts/src/v0.8/automation/dev/test/AutomationRegistry2_2.t.sol:1]: Sure, will do. @FelixFan1992 can you elaborate a bit more on `someone sends a eth_call`? Trying to see what is the best way to cover this scenario. Thanks.

Comment (FelixFan1992) [reply contracts/src/v0.8/automation/dev/test/AutomationRegistry2_2.t.sol:1]: the reason we add this new address is to restrict which address can call `checkUpkeep`, `simulatePerformUpkeep` etc. only `eth_call` from this address is allowed. so we can add a simple testcase for this. maybe just for `checkUpkeep`

File: contracts/src/v0.8/automation/dev/test/AutomationRegistry2_2.t.sol [hunk 2]
```diff
@@ -85,6 +88,22 @@ contract AutomationRegistry2_2_SetConfig is AutomationRegistry2_2_SetUp {
...
+
+    uint256 id = 1;
+    bytes memory triggerData = abi.encodePacked("trigger_data");
+
+    (bool success, ) = address(registry).staticcall(
+      abi.encodeWithSelector(AutomationCompatibleInterface.checkUpkeep.selector, id, triggerData)
+    );
+
+    // Expect the call to revert with "OnlySimulatedBackend" error
+    require(!success, "Expected eth_call to revert for not allowed address");
```
Comment (shileiwill) [inline contracts/src/v0.8/automation/dev/test/AutomationRegistry2_2.t.sol:104]: Added this test. Not sure if I am doing it the right way. Looking for comments.

**Review by RyanRHall — APPROVED** [review]

**Pull request merged**